docs/topics/queues.rst

```diff
@@ -0,0 +1,84 @@
+.. _topics-queues:
+
+======
+Queues
+======
+
+Scrapy uses queues to schedule requests. By default, a memory-based queue is
+used but if :ref:`crawls should be paused and resumed <topics-jobs>` or more
+requests than fit in memory are scheduled, using an external queue (disk
+queue) is necessary. The setting :setting:`SCHEDULER_DISK_QUEUE` determines
+the type of disk queue that will be used by the scheduler.
```

Comment:
Even though you are creating this page with external queues in mind, this is the page where users interested in finding general information about Scrapy request queuing system will go. So I think we need to extend this documentation to cover questions they may have, even if they are unrelated to external queues.

For example:
- Why does Scrapy use multiple queues, and not a single one?
- How does Scrapy determine that additional requests do not fit into memory? How much margin of error is there? Is there a setting to adjust this? Where does Scrapy store requests in this scenario, where `JOBDIR` has not been defined?

Also:
- `JOBDIR` could be used just to save memory, even if no pausing and resuming is desired, so it’s probably better to mention the setting being used and not the pausing and resuming
- Maybe this page should be part of the “Extending Scrapy” section, and be linked from somewhere in the “Architecture overview” page.

Reply:
> > Why does Scrapy use multiple queues, and not a single one?
> 
> Do you mean memory vs. disk queue here, or priority queues?

I meant priorities (you already covered the disk vs memory aspect).

> > Maybe this page should be part of the “Extending Scrapy” section
> 
> It's already part of the `Extending scrapy` section in `index.rst`. Or do you mean something else?

Oops, my bad :facepalm:. For some reason, after reading the reference to the topic covering JOBDIR, I just assumed it was placed after that topic. Forget that point.

Reply:
@Gallaecio I extended the documentation with information about priority queues and refined the part about downstream queues. Maybe you want to take a look.